doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,213 @@
+transformed into ``/*!50633 COLUMN_FORMAT COMPRESSED */``. In this mode no
+``CREATE/DROP COMPRESSION_DICTIONARY ...`` statements will ever be generated.
+
+When a new :option:`enable-compressed-columns-with-dictionaries`` option is
+specified, all compressed column attributes and compression dictionary
+references will be preserved.
+Moreover, the following dictionary creation fragments will be added before
+``CREATE TABLE`` statements which are going to use these dictionaries for the
+first time.
+
+.. code-block:: mysql
+
+  /*!50633 DROP COMPRESSION_DICTIONARY IF EXISTS <dictionary>; */
+  /*!50633 CREATE COMPRESSION_DICTIONARY <dictionary>(...); */
+
+Two new options :option:`add-drop-compression-dictionary` and
+:option:`skip-add-drop-compression-dictionary` will control if ``/*!50633 DROP
+COMPRESSION_DICTIONARY IF EXISTS <dictionary> */`` part from previous paragraph
+will be skipped or not. By default, :option:`add-drop-compression-dictionary`
+mode will be used.
+
+When both :option:`enable-compressed-columns-with-dictionaries` and
+``--tab=<dir>`` (separate file for each table) options are specified, necessary
+compression dictionaries will be created in each output file using the
+following fragment (regardless of the values of
+:option:`add-drop-compression-dictionary` and
+:option:`skip-add-drop-compression-dictionary` options).
+
+.. code-block:: mysql
+
+  /*!50633 CREATE COMPRESSION_DICTIONARY IF EXISTS <dictionary>(...); */
+
+Downgrade scenario
+==================
+
+If it is necessary to perform |Percona Server| downgrade from a version
+:rn:`5.6.33-79.0` (or newer) to a version older than :rn:`5.6.33-79.0` and if
+user databases have one or more table with compressed columns, there are two
+options to do this safely:
+
+1. Use ``mysqldump`` in compatible mode (no compressed columns extensions must
+   be specified).
+
+2. Manually remove the ``COMPRESSED`` attribute from all columns which have it
+   via ``ALTER TABLE ... MODIFY ... COLUMN_FORMAT DEFAULT`` before updating
+   server binaries.
+   In this case, the downgraded server can start safely with old data files.
+
+Version Specific Information
+============================
+
+  * :rn:`5.6.33-79.0`
+    Feature implemented in |Percona Server| 5.6
+
+System Variables
+================
+
+.. variable:: innodb_compressed_columns_zip_level
+
+   :cli: Yes
+   :conf: Yes
+   :scope: Global
+   :dyn: Yes
+   :vartype: Numeric
+   :default: 6
+   :range: ``0``-``9``
+
+This variable is used to specify the compression level used for compressed
+columns. Specifying ``0`` will use no compression, ``1`` the fastest and ``9``
+the best compression. Default value is ``6``.
+
+.. variable:: innodb_compressed_columns_threshold
+
+   :cli: Yes
+   :conf: Yes
+   :scope: Global
+   :dyn: Yes
+   :vartype: Numeric
+   :default: 96
+   :range: ``1`` - ``2^64-1`` (or ``2^32-1`` for 32-bit release)
+
+This variable is used to specify the column length after which the column will
```

Comment:
This variable is used to specify minimal length of the inserted value after which the compression algorithm will be applied. Setting it properly can prevent situations when compressed representation of a very small byte sequence is longer than in the original value.